fixed build on Windows, added HEX string

promd opened the pull request:
Hi,
I made 2 small modifications others might benefit from:
- added CHAR_TYPE_HEX to support Hexstrings on Set
       e.g. a "Set" message of FF,01 will now work to write 255 and 01
- changed gzip to tar to ease build on Windows
       gzip is not available on Windows, while tar is

shmuelzon commented on main/CMakeLists.txt:
```diff
@@ -20,7 +20,7 @@ foreach(file ${www_files})
     file(MAKE_DIRECTORY ${www_file_dir})
     add_custom_command(OUTPUT ${build_dir}/www/${file}.gz
-        COMMAND gzip -c ${PROJECT_DIR}/www/${file} > ${build_dir}/www/${file}.gz
+        COMMAND tar -czf ${build_dir}/www/${file}.gz ${PROJECT_DIR}/www/${file}
```
Hey, does this work with the web interface? A gzipped tar isn't the same as a gzipped file as it includes some tar headers which shouldn't be sent as a response by the web server.

promd replied:
Ah, you are right, it does not work. I did not try the web frontend before, was just happy I got it compiling on windows... 🤔

shmuelzon replied:
I was looking for alternatives at the time. Particularly, a Python gzip module that seems to already be installed with ESP-IDF, but that didn't have the option of specifying the output file, or something like that

shmuelzon replied:
Actually, looking at the code of that module now, you might be able to get away with:
```cmake
COMMAND ${python} -m gzip < ${PROJECT_DIR}/www/${file} > ${build_dir}/www/${file}.gz`
```
If it does work, I'd appreciate a separate pull request for it or, at least, a separate commit.
Thanks!

promd replied:
I'll give it a try and return with an extra PR. 
Have some other changes anyways.

Thanks for the cool Project !!

promd replied:
Still produces the same error: :-/

httpd_txrx: httpd_resp_send_err: 431 Request Header Fields Too Large - Header fields are too long for server to interpret

promd commented:
I modified the existing PR, I think... still an old SVN user.
A new parameter has been added, b/c I was running into issues with a characteristic NOTIFY.
I also reverted back to the GZIP command....
